extensions/blocks/podcast-player/podcast-player.js

```diff
@@ -0,0 +1,372 @@
+/* global _wpmejsSettings, MediaElementPlayer */
+/**
+ * Internal dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import debugFactory from 'debug';
+import { soundIcon, errorIcon } from './icons';
+
+const debug = debugFactory( 'jetpack:podcast-player' );
+
+const meJsSettings = typeof _wpmejsSettings !== 'undefined' ? _wpmejsSettings : {};
+
+const STATE_PLAYING = 'is-playing';
+const STATE_ERROR = 'is-error';
+const STATE_PAUSED = 'is-paused';
+
+const noop = function() {};
+
+/** Class Podcast Player */
+class PodcastPlayer {
+	playerState = null;
+	id = null;
+	block = null;
+	currentTrack = -1;
+	tracks = [];
+	attributes = {};
+	audio = null;
+	mediaElement = null;
+	trackElements = [];
+
+	/**
+	 * Create player instance.
+	 * @param {string} id The id of the block element in document.
+	 */
+	constructor( id ) {
+		// Find DOM node.
+		const block = document.getElementById( id );
+		debug( 'constructing', id, block );
+
+		// Check if we can find the block and required dependency.
+		if ( ! block || ! MediaElementPlayer ) {
+			return;
+		}
+
+		// Create player instance.
+		this.id = id;
+		this.block = block;
+
+		// Load data from the embedded JSON.
+		this.loadMetadata();
+		if ( ! this.tracks.length ) {
+			return;
+		}
+
+		// Initialize player UI.
+		this.initializeAudio();
+
+		// Store track elements and cast as Array.
+		this.trackElements = [ ...block.querySelector( '.podcast-player__episodes' ).children ];
+
+		// Attach event handlers.
+		this.block.addEventListener( 'click', this.handleClick.bind( this ), false );
+		this.block.addEventListener( 'keydown', this.handleKeyDown.bind( this ), false );
+	}
+
+	/**
+	 * Lookup JSON data in block and load them into this intance.
+	 * @private
+	 */
+	loadMetadata() {
```

Comment:
Is there any Jetpack convention for labelling private labels with an underscore or the like? i.e. `_loadMetadata()`?

Reply:
Not aware of any convention. On the PHP side, there is obviously native support for private functions but in JS, not really